ts/features/idpay/configuration/screens/IdPayDiscountInstrumentsScreen.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  RouteProp,
+  useFocusEffect,
+  useNavigation,
+  useRoute
+} from "@react-navigation/native";
+import React from "react";
+import { H1, VSpacer } from "@pagopa/io-app-design-system";
+import * as pot from "@pagopa/ts-commons/lib/pot";
+
+import { ScrollView, StyleSheet } from "react-native";
+import LoadingSpinnerOverlay from "../../../../components/LoadingSpinnerOverlay";
+import { Body } from "../../../../components/core/typography/Body";
+import I18n from "../../../../i18n";
+import customVariables from "../../../../theme/variables";
+import { emptyContextualHelp } from "../../../../utils/emptyContextualHelp";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import {
+  idPayInitiativesInstrumentRefreshStart,
+  idPayInitiativesInstrumentRefreshStop,
+  idpayInitiativeInstrumentDelete
+} from "../store/actions";
+import {
+  idPayIsLoadingInitiativeInstrumentSelector,
+  idpayDiscountInitiativeInstrumentsSelector,
+  isLoadingDiscountInitiativeInstrumentsSelector
+} from "../store";
+import { IdPayDiscountInstrumentEnrollmentSwitch } from "../components/IdPayDiscountInstrumentEnrollmentSwitch";
+import { IDPayConfigurationParamsList } from "../navigation/navigator";
+import TopScreenComponent from "../../../../components/screens/TopScreenComponent";
+import { useIdPayInfoCieBottomSheet } from "../../code/components/IdPayInfoCieBottomSheet";
+import { InstrumentTypeEnum } from "../../../../../definitions/idpay/InstrumentDTO";
+import { IdPayCodeRoutes } from "../../code/navigation/routes";
+import { IOStackNavigationProp } from "../../../../navigation/params/AppParamsList";
+import { IdPayCodeParamsList } from "../../code/navigation/params";
+
+type IdPayDiscountInstrumentsScreenRouteParams = {
+  initiativeId: string;
+  initiativeName?: string;
+};
+
+type IdPayDiscountInstrumentsScreenRouteProps = RouteProp<
+  IDPayConfigurationParamsList,
+  "IDPAY_CONFIGURATION_DISCOUNT_INSTRUMENTS"
+>;
+
+/**
+ * Screen that shows the list of available instruments for a discount initiative which has been selected
+ * Actually are available only the CIE and the QRCode
+ */
+const IdPayDiscountInstrumentsScreen = () => {
+  const dispatch = useIODispatch();
+  const route = useRoute<IdPayDiscountInstrumentsScreenRouteProps>();
+  const navigation =
+    useNavigation<IOStackNavigationProp<IdPayCodeParamsList>>();
+  const { initiativeId, initiativeName } = route.params;
+
+  const initiativeInstruments = useIOSelector(
+    idpayDiscountInitiativeInstrumentsSelector
+  );
+  const isLoadingInstruments = useIOSelector(
+    isLoadingDiscountInitiativeInstrumentsSelector
+  );
+
+  const idPayCodeInitiative = React.useMemo(
+    () =>
+      initiativeInstruments.find(
+        initiative => initiative.instrumentType === InstrumentTypeEnum.IDPAYCODE
+      ),
+    [initiativeInstruments]
+  );
```

Comment:
Just to be clear, this object is an instrument and not an initiative. Object names should be self-esplicative.
```suggestion
  const idPayCodeInstrument = React.useMemo(
    () =>
      initiativeInstruments.find(
        initiative => initiative.instrumentType === InstrumentTypeEnum.IDPAYCODE
      ),
    [initiativeInstruments]
  );
```

Reply:
Addressed into 88119f5289e559fb5bbe5b9e3900263d585df0ce